src/physics/em/BetheHeitlerInteractor.i.hh

```diff
@@ -0,0 +1,229 @@
+                // Calculate delta from element atomic number and sampled
+                // epsilon
+                real_type delta = this->impact_parameter(epsilon);
+                CHECK(delta <= delta_max && delta >= delta_min);
+                // Calculate g1 "rejection" function
+                reject_threshold
+                    = celeritas::max(this->screening_phi1_aux(delta), 0.0)
+                      / celeritas::max(this->screening_phi1_aux(delta_min),
+                                       0.0);
+                CHECK(reject_threshold > 0.0 && reject_threshold <= 1.0);
+            }
+            else
+            {
+                // Used to sample from f2
+                epsilon = epsilon_min
+                          + (0.5 - epsilon_min) * generate_canonical(rng);
+                CHECK(epsilon >= epsilon_min && epsilon <= 0.5);
+                // Calculate delta given the element atomic number and sampled
+                // epsilon
+                real_type delta = this->impact_parameter(epsilon);
+                CHECK(delta <= delta_max && delta >= delta_min);
+                // Calculate g2 "rejection" function
+                reject_threshold
+                    = celeritas::max(this->screening_phi2_aux(delta), 0.0)
+                      / celeritas::max(this->screening_phi2_aux(delta_min),
+                                       0.0);
+                CHECK(reject_threshold >= 0.0 && reject_threshold <= 1.0);
+            }
+        } while (BernoulliDistribution(1.0 - reject_threshold)(rng));
+    }
+
+    // Select charges randomly
+    real_type totalE_ele;
+    real_type totalE_pos;
+    if (BernoulliDistribution(0.5)(rng))
+    {
+        totalE_ele = (1.0 - epsilon) * inc_energy_.value();
+        totalE_pos = epsilon * inc_energy_.value();
+    }
+    else
+    {
+        totalE_ele = epsilon * inc_energy_.value();
+        totalE_pos = (1.0 - epsilon) * inc_energy_.value();
+    }
+
+    // Sample pair kinematics
+    secondaries[0].energy = units::MevEnergy{
+        celeritas::max(0.0, totalE_ele - 1.0 / shared_.inv_electron_mass)};
+    secondaries[1].energy = units::MevEnergy{
+        celeritas::max(0.0, totalE_pos - 1.0 / shared_.inv_electron_mass)};
+
+    // Sample pair directions
+    for (int i = 0; i < 2; ++i)
+    {
+        real_type phi = 2 * constants::pi * generate_canonical(rng);
+        real_type umax
+            = 2
+              * (1.0
+                 + secondaries[i].energy.value() / shared_.inv_electron_mass);
+        real_type u;
+        do
+        {
+            real_type uu
+                = -std::log(generate_canonical(rng) * generate_canonical(rng));
+            u = BernoulliDistribution(0.25)(rng) ? uu * 1.6
+                                                 : uu * (1.0 / 1.875);
+        } while (u > umax);
+
+        real_type cost = std::pow(-1, i)
+                         * (1.0 - 2.0 * ipow<2>(u) / ipow<2>(umax));
```

Comment:
Instead of `std::pow(-1, i)` (which is mathy but not computer-friendly) use an expression that's more readable ("-1 if electron, 1 if positron"); also you can squish the ratio inside the `ipow`:
```suggestion
        real_type cost = (i == 0 ? -1 : 1) * (1 - 2 * ipow<2>(u / umax));
```